### PR TITLE
Implement `test:xs` for `vats` package

### PR DESCRIPTION
**Pull request by michaelfig**
## Description

Testing `vats` upgrade and bootstrap under both `local` and `xs-worker` systems helps our code be more robust.  This change enables those tests via our existing CI, by implementing `yarn test:xs` in `packages/vats/package.json`.

Some adjustments to the bootstrap tests were needed to accommodate differences in the environments (they were sometimes too specific to just the Node.js local worker).

### Security Considerations

n/a

### Scaling Considerations

Improves developer experience by speeding up `yarn test` for this package.

### Documentation Considerations

n/a

### Testing Considerations

This change should not negatively impact the CI performance, as XS and Node.js tests are already run in parallel, and XS is bypassed for most packages in the CI jobs.

### CODE REVIEW

File: packages/vats/test/bootstrapTests/test-vaults-integration.js [hunk 1]
```diff
@@ -167,12 +167,12 @@ test('close vault', async t => {
...
     ),
     {
-      message: /^Withdrawal .* failed because the purse only contained .*/,
+      message: /^Withdrawal of .* failed because the purse only contained .*/,
     },
   );
 
   const message =
-    'Offer {"brand":"[Alleged: IST brand]","value":"[1n]"} is not sufficient to pay off debt {"brand":"[Alleged: IST brand]","value":"[5025000n]"}';
+    /Offer (\(an object\)|{"brand":"\[Alleged: IST brand\]","value":"\[1n\]"}) is not sufficient to pay off debt (\(an object\)|{"brand":"\[Alleged: IST brand\]","value":"\[5025000n\]"})$/;
```
Comment (dckc) [inline packages/vats/test/bootstrapTests/test-vaults-integration.js]: I wonder if this points to a missing `q(amount)` somewhere.

File: packages/vats/test/upgrading/test-upgrade-contracts.js [hunk 2]
```diff
@@ -32,7 +32,7 @@ test('upgrade mintHolder', async t => {
 
   /** @type {SwingSetConfig} */
   const config = harden({
-    defaultManagerType: 'xs-worker',
+    defaultManagerType: 'local', // Overridden in CI with SWINGSET_WORKER_TYPE=xs-worker
```
Comment (dckc) [inline packages/vats/test/upgrading/test-upgrade-contracts.js:35]: relying on an environment variable to override this seems like spooky-action-at-a-distance.

But... I don't suppose I have a better idea.

**Review by dckc — APPROVED** [review]
looks good.

cc @gibson042

File: packages/vats/test/bootstrapTests/test-vaults-integration.js (hunk 1 above)
Comment (michaelfig) [reply packages/vats/test/bootstrapTests/test-vaults-integration.js]: Actually, it's an indicator that `@endo/ses-ava` can't unhide details when the vats are in a separate agent from the kernel.  There's not really a way to both allow censorship of important private stuff in production and also have insight into the censored stuff from another process during debugging.

File: packages/vats/test/upgrading/test-upgrade-contracts.js (hunk 2 above)
Comment (michaelfig) [reply packages/vats/test/upgrading/test-upgrade-contracts.js:35]: That's the way all the other `test:xs` scripts are structured, so at least there's a precedent.

File: packages/vats/test/bootstrapTests/test-vaults-integration.js (hunk 1 above)
Comment (dckc) [reply packages/vats/test/bootstrapTests/test-vaults-integration.js]: What I meant is: maybe this amount shouldn't be considered sensitive.

File: packages/vats/test/bootstrapTests/supports.js [hunk 3]
```diff
@@ -26,6 +26,29 @@ export const bootstrapMethods = {
...
+ * @template V
+ * @param {K[]} keys
+ * @param {(key: K, i: number) => V} valueMaker
+ */
+const keysToObject = (keys, valueMaker) => {
+  return Object.fromEntries(keys.map((key, i) => [key, valueMaker(key, i)]));
+};
+
+/**
+ * AVA's default t.deepEqual() is nearly unreadable for sorted arrays of strings.
```
Comment (turadg) [inline packages/vats/test/bootstrapTests/supports.js:41]: 👏 I was just thinking yesterday of whether to propose that marcher for Ava. I bet they'd say to keep it in a separate package. Fortunately they're considering a custom marchers mechanism to add ones to `t`.

File: packages/vats/test/bootstrapTests/test-vaults-integration.js (hunk 1 above)
Comment (turadg) [reply packages/vats/test/bootstrapTests/test-vaults-integration.js]: Agree that the amounts shouldn't be sensible. Please quote it.

File: packages/vats/test/bootstrapTests/test-vaults-integration.js [hunk 4]
```diff
@@ -187,14 +187,20 @@ test('close vault', async t => {
...
+  const updateRecord = wd.getLatestUpdateRecord();
+  t.like(updateRecord, {
     updated: 'offerStatus',
     status: {
       id: 'close-insufficient',
       numWantsSatisfied: 1, // trivially true because proposal `want` was empty.
-      error: `Error: ${message}`,
     },
   });
+  if (updateRecord.updated === 'offerStatus') {
```
Comment (turadg) [inline packages/vats/test/bootstrapTests/test-vaults-integration.js]: This condition is for type narrowing? Worth a comment. 

But with a quoted value could the simpler like remain?

File: scripts/configure-vscode.sh [hunk 5]
```diff
@@ -23,6 +23,7 @@ cat >.vscode/settings.json.new <<\EOF ||
   "editor.defaultFormatter": "esbenp.prettier-vscode",
 
   "typescript.preferences.importModuleSpecifierEnding": "js",
+  "typescript.tsdk": "node_modules/typescript/lib",
```
Comment (turadg) [inline scripts/configure-vscode.sh:26]: What difference does this make? Was your VSC version too far ahead?

File: packages/vats/test/upgrading/test-upgrade-vats.js [hunk 6]
```diff
@@ -38,7 +38,7 @@ const makeScenario = async (
   /** @type {SwingSetConfig} */
   const config = {
     includeDevDependencies: true, // for vat-data
-    defaultManagerType: 'xs-worker',
+    defaultManagerType: 'local', // Overridden in CI with SWINGSET_WORKER_TYPE=xs-worker
```
Comment (turadg) [inline packages/vats/test/upgrading/test-upgrade-vats.js:41]: Out of scope but it looks like DRYing up some of these config cases is worthwhile

File: packages/vats/test/bootstrapTests/test-vaults-integration.js (hunk 4 above)
Comment (michaelfig) [reply packages/vats/test/bootstrapTests/test-vaults-integration.js]: Yes, the quoted value would make this unnecessary.

File: scripts/configure-vscode.sh (hunk 5 above)
Comment (michaelfig) [reply scripts/configure-vscode.sh:26]: I just remember needing to advise some folks to "Select Typescript Version" to point to the SDK repo in the past in order to fix discrepancies between VSCode and `tsc`.

**Review by turadg — APPROVED** [review]

**Pull request merged**